### ISSUE TITLE
Add a dummy __call__ method to nn.Module base class for typechecking

### ISSUE THREAD

**Issue by levskaya** [google/flax#1364]
for pytype and other checkers we should have something like:

```
if typing.TYPE_CHECKING:
  def __call__(self, *args, **kwargs) -> Any: ...
```

at the very least to remove a class of existing false-negative type-check issues in pytype (when passing modules / module factories as attributes).

**Comment by levskaya** [comment]
A minimal example of one current issue w. the pytype checker is:
```python
from flax import linen as nn
class MyModule(nn.Module):
  sub_module: nn.Module
  def __call__(self):
    return self.sub_module()
```
results in `... in __call__: 'Module' object is not callable [not-callable]`

**Comment by levskaya** [comment]
Closing - this was resolved by #1365
